### PR TITLE
ISSUE-601 [Public agenda] rest apis to patch and delete booking links

### PR DESCRIPTION
**Pull request by hungphan227**
## Summary by CodeRabbit

* **New Features**
  * DELETE and PATCH endpoints for booking links (delete returns 204; patch supports active, duration, calendar URL, and availabilityRules with strict validation and time-zone handling).

* **Documentation**
  * Added comprehensive Booking Link API docs with schemas, examples, and error details.

* **Tests**
  * Added extensive integration tests for DELETE/PATCH flows, authentication, validation, edge cases, and minor test assertion fixes.

* **Refactor**
  * Introduced a dedicated AvailabilityRule DTO and an AvailabilityRuleType enum; create/get routes now use the shared DTO.

### CODE REVIEW

File: calendar-rest-api/src/main/java/com/linagora/calendar/restapi/routes/BookingLinkPatchRoute.java [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+                parseAvailabilityRules(node));
+        } catch (IllegalArgumentException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Invalid request body", e);
+        }
+    }
+
+    private ValuePatch<CalendarURL> parseCalendarUrl(JsonNode node) {
+        if (!node.has(FIELD_CALENDAR_URL)) {
```
Comment (vttranlina) [inline calendar-rest-api/src/main/java/com/linagora/calendar/restapi/routes/BookingLinkPatchRoute.java]: Currently, the `BookingLinkPatchRequest` does not accept 
```
        Preconditions.checkArgument(!calendarUrl.isRemoved(), "'calendarUrl' can not be removed");
        Preconditions.checkArgument(!duration.isRemoved(), "'eventDuration' can not be removed");
        Preconditions.checkArgument(!active.isRemoved(), "'active' can not be removed");
```
We can trust it

**Review by vttranlina — APPROVED** [review]

File: docs/apis/bookingLink.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+
+**Request body**
+
+| Field               | Required | Description                                        |
+|---------------------|----------|----------------------------------------------------|
+| `calendarUrl`       | yes      | Calendar URI (must be accessible by the user)      |
+| `durationMinutes`   | yes      | Slot duration in minutes, must be positive         |
+| `active`            | yes      | Whether the booking link is active                 |
+| `timeZone`          | no       | Defaults to `UTC` if omitted                       |
+| `availabilityRules` | no       | List of availability rules                         |
```
[the remark this answers is not in the document]
Comment (hungphan227) [reply docs/apis/bookingLink.md]: it is supposed to be default business hours. will be updated in https://github.com/linagora/twake-calendar-side-service/issues/625
